File: CodingChallenge/ArrowFunctionObjectDestructuring.dart
void main() {
  // Function to square a number using an anonymous arrow function
  int square(int x) => x * x;

  print(square(5)); // Output: 25

  // Map representing a user with properties
  var user = {
    'firstName': 'Raphael',
    'lastName': 'Africano',
    'age': 29,
  };

  // Extract properties with type annotations
  String firstName = user['firstName'] as String;
  String lastName = user['lastName'] as String;
  int age = user['age'] as int;

  print(firstName); // Output: Raphael
  print(lastName); // Output: Africano
  print(age); // Output: 29

  // Define a class representing a person with a greet method
  var person = Person(
    firstName: firstName,
    lastName: lastName,
    age: age,
  );

  print(person.firstName); // Output: Raphael
  print(person.age); // Output: 29
  print(person.greet()); // Call 'greet' method using parentheses to execute it
  // Output: Hello, my name is Raphael Africano.
}

// Define a class to represent a person with properties and methods
class Person {
  final String firstName;
  final String lastName;
  final int age;

  // Constructor for initializing the person object
  Person({
    required this.firstName,
    required this.lastName,
    required this.age,
  });

  // Method to greet the person
  String greet() {
    return 'Hello, my name is $firstName $lastName.';
  }
}

File: CodingChallenge/ArrowFunctionObjectDestructuring.php
<?php
// Function to square a number
function square($x) {
    return $x * $x;
}

echo square(5) . "\n"; // Output: 25

// Array representing a user with properties
$user = [
    'firstName' => 'Raphael',
    'lastName' => 'Africano',
    'age' => 29
];

// Extract properties from the user array
$firstName = $user['firstName'];
$lastName = $user['lastName'];
$age = $user['age'];

echo $firstName . "\n"; // Output: Raphael
echo $lastName . "\n";  // Output: Africano
echo $age . "\n";       // Output: 29

// Define a class representing a person with properties and a greet method
class Person {
    public $firstName;
    public $lastName;
    public $age;

    // Constructor for initializing the person object
    public function __construct($firstName, $lastName, $age) {
        $this->firstName = $firstName;
        $this->lastName = $lastName;
        $this->age = $age;
    }

    // Method to greet the person
    public function greet() {
        return "Hello, my name is $this->firstName $this->lastName.";
    }
}

// Create a new Person object
$person = new Person($firstName, $lastName, $age);

echo $person->firstName . "<br>\n"; // Output: Raphael
echo $person->age . "<br>\n";       // Output: 29
echo $person->greet() . "<br>\n";   // Output: Hello, my name is Raphael Africano.
?>
